qiskit/optimization/problems/quadratic_program.py

```diff
@@ -1092,6 +1093,63 @@ def from_ising(self,
         self.minimize(constant=offset, linear=linear_terms, quadratic=quadratic_terms)
         offset -= offset
 
+    def is_feasible(self, x: Union[List[float], np.ndarray], detailed: bool = False) -> bool:
+        """Returns whether a solution provided by the optimizer is feasible or not
+        Args:
+            x: the input result list returned by the optimizer
+            detailed: whether or not to print the violations
+        Returns:
+            is_feasible: Whether the solution provided by the optimizer is feasible or not.
+
+        Raises:
+            QiskitOptimizationError: If the input `x` is not same len as total vars
+        """
+        # if input `x` is not the same len as the total vars, raise an error
+        if len(x) != self.get_num_vars():
+            raise QiskitOptimizationError(
+                'The size of `x` differs from the total number of variables.'
+                ' size of `x`: {}, num. of vars: {}'.format(len(x), self.get_num_vars())
+            )
+
+        # check whether the input satisfy the bounds of the problem
+        satisfied_variables = {}
+        for i, val in enumerate(x):
+            variable = self.get_variable(i)
+            if variable._lowerbound <= val <= variable._upperbound:
+                satisfied_variables[variable.name] = True
+                # print(f'{variable._name} is within the bounds')
```

Comment:
The logger will only print the information when the logger is enabled at the specific level - so if this were logged at DEBUG level it would not get written by default, only if log level is set to include DEBUG info. If some return structure around variables/constraints ends up getting passed back too maybe any printing of this is no so useful. In the end it comes down to whether its felt that having this print out might be helpful in debugging an issues that might arise. It is something we can ask a user to turn on so when they run we get such log information output which they can show us to help us understand what was happening if say we cannot produce the problem locally ourselves when debugging it but the user can. I cannot say how helpful this might be - if its seems not so useful then by all means remove it completely.

Reply:
If it can be used for DEBUG purposes, would be good to just log the whole dictionary which contains the info of all variables/constraints and whether they passed the check or not?

Reply:
The logger can be used to log whatever might be thought useful to aid debugging for the purposes I described. I'll leave it to others that know the code more intimately than me to say whether they think its useful. I just do not want to end up with print statements in code, whether they are commented out or not.

Reply:
Absolutely, I will just log the violations and commit it and let the reviewers decide what is required and what it not.